src/widgets/helper/ChannelView.cpp

```diff
@@ -2134,6 +2196,36 @@ void ChannelView::addImageContextMenuItems(
             addEmoteContextMenuItems(*emoteElement->getEmote(), creatorFlags,
                                      menu);
         }
+        else if (auto layeredElement =
+                     dynamic_cast<const LayeredEmoteElement *>(&creator))
+        {
+            // Functor for std::copy_if that keeps track of seen elements
```

Comment:
I'd prefer if "get unique emotes" is added as a helper function to LayeredEmoteElement instead